citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/CitationManager.java

```diff
@@ -494,4 +496,28 @@ public CitationPage getPrevAndNextCitation(IUser user, String groupId, String co
     public void deleteLocalGroupCitations(String groupId) {
         citationStore.deleteCitationByGroupId(groupId);
     }
+
+    @Override
+    public ICitation addCitationToReferences(IUser user, ICitation citation, String zoteroGroupId, 
+            String referenceCitationKey, String reference) throws SelfCitationException, 
+            ZoteroConnectionException, CitationIsOutdatedException, ZoteroHttpStatusException, 
+            ZoteroItemCreationFailedException {
+        Set<IReference> references = citation.getReferences();
+        if (references == null) {
+            references = new HashSet<>();
+            citation.setReferences(references);
+        }
+        if (!citation.getKey().equals(referenceCitationKey)) {
+            IReference newReference = new Reference();
+            newReference.setCitationId(referenceCitationKey);
+            newReference.setReferenceString(reference);
+            references.add(newReference);
+        } else {
```

Comment:
let's turn this if/else around so that we loose one indented block, so
```
if keys are equal:
      throw exception
code if they are not equal goes here
```